pkg/controller/enterprisesearch/config.go

```diff
@@ -81,9 +86,42 @@ func ReconcileConfig(client k8s.Client, ents entsv1beta1.EnterpriseSearch) (*cor
 	return reconciledConfigSecret, nil
 }
 
+// reusableSettings captures secrets settings in the Enterprise Search configuration that we want to reuse.
+type reusableSettings struct {
+	SecretSession  string   `config:"secret_session_key"`
+	EncryptionKeys []string `config:"secret_management.encryption_keys"`
+}
+
+// reuseOrGenerateSecretKeys reads the current configuration and reuse existing secrets it they exist.
+func reuseOrGenerateSecretKeys(c k8s.Client, ents entsv1beta1.EnterpriseSearch) (*settings.CanonicalConfig, error) {
+	cfg, err := getExistingConfig(c, ents)
+	if err != nil {
+		return nil, err
+	}
+
+	var e reusableSettings
+	if cfg == nil {
+		e = reusableSettings{}
+	} else if err := cfg.Unpack(&e); err != nil {
+		return nil, err
+	}
+	if len(e.SecretSession) == 0 {
+		e.SecretSession = rand.String(32)
+	}
+	if len(e.EncryptionKeys) == 0 {
+		e.EncryptionKeys = []string{rand.String(32)}
+	}
+	return settings.MustCanonicalConfig(e), nil
+}
+
 func newConfig(c k8s.Client, ents entsv1beta1.EnterpriseSearch) (*settings.CanonicalConfig, error) {
 	cfg := defaultConfig(ents)
 
+	secretCfg, err := reuseOrGenerateSecretKeys(c, ents)
```

Comment:
suggestion: `reusedCfg, err := getOrCreateReusableSettings()` ?
(secretCfg is a bit ambiguous)